### PR TITLE
Add data presevation

### PR DESCRIPTION
**Pull request by Thinus01**
### General requirements

- Made sure that there are no linter errors.
- Made sure that I used correct flow Gitflow.
- Made sure that I documented my work in a professional way.

### Create custom methods
I included additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Thinus01 👋,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Status: Invalid Code Review Request



You have submitted a project, that used the wrong flow. Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by GzimAsani — CHANGES_REQUESTED** [review]
Hi @Thinus01  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that did not use Microverse Readme [template](https://github.com/microverseinc/readme-template), kindly use it and personalize the headers and content 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Thinus01** [comment]
Hi @GzimAsani 

My README is based on that template.

I have had reviewers tell me to remove that logo from the top so that's why it's not included.
The table of content is a drop-down, however, it still contains all the relevant links from the template.

**Comment by GzimAsani** [comment]
> Hi @GzimAsani
> 
> My README is based on that template.
> 
> I have had reviewers tell me to remove that logo from the top so that's why it's not included. The table of content is a drop-down, however, it still contains all the relevant links from the template.

Yes, sorry I missed this dropdown, I didn't see the menu that's why
image

**Comment by Thinus01** [comment]
No problem, invalid code reviews at least do not cost me anything.

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_save :update_counter
+
+  private
+
+  def update_counter
+    post.increment!(like_counter)
```
Comment (GzimAsani) [inline app/models/like.rb]: - [ ] You have a typo here,  I think this needs to be `post.increment!(:likes_counter)`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Thinus01                 :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with models, they meet the requirements :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Thinus01, 
congratulations-1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Change :heavy_check_mark: 
Created Models Correctly :heavy_check_mark: 
Good Commit Message :heavy_check_mark: 
Descriptive Branch Name :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**